### ISSUE TITLE
[BUG] DataFrame.loc fails on DateTime index

### ISSUE THREAD

**Issue by miguelusque** [rapidsai/cudf#5328]
**Describe the bug**
Hi!

I am facing an error when trying to use `DataFrame.loc` on a `DateTime` index:

```
import cudf
import cupy

random_df = cudf.DataFrame.from_records(cupy.random.randn(6, 4), columns=list('ABCD'))

dates = cudf.Series(['2021-01-01', '2021-01-02', '2021-01-03',
                     '2021-01-04', '2021-01-05', '2021-01-06'], dtype='datetime64[s]')

random_df = random_df.set_index(dates)

random_df.loc[dates[0]]
```

Output:

```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-3-f970050aac9c> in <module>
----> 1 random_df.loc[dates[0]]

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/indexing.py in __getitem__(self, arg)
    114             if not isinstance(arg, tuple):
    115                 arg = (arg, slice(None))
--> 116             return self._getitem_tuple_arg(arg)
    117 
    118     def __setitem__(self, key, value):

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/indexing.py in _getitem_tuple_arg(self, arg)
    217             for col in columns_df.columns:
    218                 # need Series() in case a scalar is returned
--> 219                 df[col] = Series(columns_df[col].loc[arg[0]])
    220             df.columns = columns_df.columns
    221 

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/indexing.py in __getitem__(self, arg)
     60     def __getitem__(self, arg):
     61         arg = self._loc_to_iloc(arg)
---> 62         return self._sr.iloc[arg]
     63 
     64     def __setitem__(self, key, value):

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/indexing.py in __getitem__(self, arg)
     42         if isinstance(arg, tuple):
     43             arg = list(arg)
---> 44         return self._sr[arg]
     45 
     46     def __setitem__(self, key, value):

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/series.py in __getitem__(self, arg)
    453 
    454     def __getitem__(self, arg):
--> 455         data = self._column[arg]
    456         index = self.index.take(arg)
    457         if is_scalar(data) or data is None:

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/column/column.py in __getitem__(self, arg)
    434         if isinstance(arg, Number):
    435             arg = int(arg)
--> 436             return self.element_indexing(arg)
    437         elif isinstance(arg, slice):
    438 

/opt/conda/envs/rapids/lib/python3.7/site-packages/cudf/core/column/column.py in element_indexing(self, index)
    414             index = len(self) + index
    415         if index > len(self) - 1:
--> 416             raise IndexError
    417 
    418         val = self[index : (index + 1)]

IndexError: 
```

**Expected behavior**
No error.

**Environment overview (please complete the following information)**
RAPIDS 0.13 on DGX-1

**Comment by rgsl888prabhu** [comment]
Seems to be fixed in 0.14, can you please try with 0.14 nightly and confirm.
```
>>> import cudf
>>> import cupy
>>> 
>>> random_df = cudf.DataFrame.from_records(cupy.random.randn(6, 4), columns=list('ABCD'))
>>> 
>>> dates = cudf.Series(['2021-01-01', '2021-01-02', '2021-01-03',
...                      '2021-01-04', '2021-01-05', '2021-01-06'], dtype='datetime64[s]')
>>> 
>>> random_df = random_df.set_index(dates)
>>> 
>>> random_df.loc[dates[0]]
A    1.517900
B    0.619596
C    1.996632
D   -1.046890
Name: 2021-01-01 00:00:00, dtype: float64

```

**Comment by miguelusque** [comment]
Hi @rgsl888prabhu ,

Thank you for your swift reply.

I have just tested it with RAPIDS 0.14 nightly build, and it seems to be working.

Closing this issue now.

Regards,
Miguel
